### PR TITLE
Feed followed by other parameter in payload

### PR DESCRIPTION
**Pull request by LianwMS**

### CODE REVIEW

**Comment by congysu** [comment]
Issue https://github.com/OData/odata.net/issues/65.

File: src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs [hunk 1]
```diff
@@ -923,7 +923,12 @@ private void ReadFeedEnd()
                 expandedNavigationLinkInfo = parentNavigationLinkScope.NavigationLinkInfo;
             }
 
-            this.jsonLightEntryAndFeedDeserializer.ReadNextLinkAnnotationAtFeedEnd(this.CurrentFeed, expandedNavigationLinkInfo, this.topLevelScope.DuplicatePropertyNamesChecker);
+            if (this.readingParameter == false)
+            {
+                // Only read annotation when not reading parameter
```
Comment (xuzhg) [inline src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs]: improve the comments.

File: test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightParameterReaderTests.cs [hunk 2]
```diff
@@ -322,6 +323,124 @@ public void ReadFeed()
         }
 
         [Fact]
+        public void ReadTwoFeeds()
```
Comment (xuzhg) [inline test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightParameterReaderTests.cs]: How about:
1. 2+ feed with difference types
2. 2+ feed type with inheritance

**Comment by xuzhg** [comment]
Others look good to me.

File: test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightParameterReaderTests.cs (hunk 2 above)
Comment (abkmr) [inline test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightParameterReaderTests.cs]: 2 feeds with different types would be an important scenario for Dynamics CRM as well.

**Comment by LianwMS** [comment]
Merge with ac4db42f994e33a9df4e2640afc42e487cd66c4a

**Pull request closed without merging**